source/grid/CMakeLists.txt

```diff
@@ -15,27 +15,49 @@
 
 INCLUDE_DIRECTORIES(BEFORE ${CMAKE_CURRENT_BINARY_DIR})
 
-SET(_src
+SET(_unity_include_src
   cell_id.cc
   grid_generator.cc
   grid_in.cc
   grid_out.cc
   grid_refinement.cc
   grid_reordering.cc
-  grid_tools.cc
   intergrid_map.cc
   manifold.cc
   manifold_lib.cc
   persistent_tria.cc
   tria_accessor.cc
   tria_boundary.cc
   tria_boundary_lib.cc
-  tria.cc
   tria_faces.cc
   tria_levels.cc
   tria_objects.cc
   )
 
+SET(_separate_src
+  grid_tools.cc
+  tria.cc
+  )
+
+# include all files in the unity file
+SET(_n_includes_per_unity_file 20)
+
+IF(DEAL_II_UNITY_BUILD)
+  SETUP_UNITY_TARGET("${_unity_include_src}" ${_n_includes_per_unity_file} _output_src)
+
+  SET(_src
+    ${_src}
+    ${_output_src}
+    ${_separate_src}
+    )
+ELSE()
+  SET(_src
+    ${_unity_include_src}
+    ${_separate_src}
+    )
+ENDIF()
```

Comment:
Just one nit-picky comment. :-)

Can we make this common idiom of 14 lines a simple macro?
Or reduce the block to 6 lines?